Next Release

Jeff-Thompson12 opened the pull request.

Jeff-Thompson12 committed:
Allow column preselection for filter inclusion

jhk0530 committed:
Update shiny_data_filter.R

replaced deprecated code with structure(NULL, *)

Jeff-Thompson12 committed:
Add basic framework

Jeff-Thompson12 committed:
Implement using an expression in numeric many

Jeff-Thompson12 committed:
Set `dev` version

Jeff-Thompson12 committed:
Merge pull request #16 from jhk0530/master

Update shiny_data_filter.R

Jeff-Thompson12 committed:
Utilize `%||%()` and set appropriate empty data structure

Jeff-Thompson12 committed:
Update NEWS

Jeff-Thompson12 committed:
Merge pull request #19 from Biogen-Inc/jt-15-NULL_cannot_have_attributes

NULL cannot have attributes

Jeff-Thompson12 committed:
Merge branch 'devel' into jt-13-preselection

Jeff-Thompson12 committed:
Update documentation

Jeff-Thompson12 committed:
Use timezone of vector instead of trying to specify

Jeff-Thompson12 committed:
Change `filter_expr` to `filter_fn`

Jeff-Thompson12 committed:
Turn back to silently failing for now

Jeff-Thompson12 committed:
Implement preselection for datetime types

Jeff-Thompson12 committed:
Update other methods with preselection

Jeff-Thompson12 committed:
Update NEWS

Jeff-Thompson12 committed:
Move preselection argument to end

Jeff-Thompson12 committed:
Make preselection reactive

Jeff-Thompson12 committed:
Make application of filter pre-selection a function

Jeff-Thompson12 committed:
Allow chaning/updating filter pre-selection

AARON-CLARK committed:
Merge pull request #21 from Biogen-Inc/jt-17-fix_shiny_vector_datetime

Use timezone of vector instead of trying to specify

Jeff-Thompson12 committed:
Resolve merge conflicts with `devel`

Merge branch 'devel' into jt-13-preselection

# Conflicts:
#	R/shiny_vector_filter_datetime.R

Jeff-Thompson12 committed:
Update documentation

Jeff-Thompson12 committed:
Add wrappers for data filter item

Jeff-Thompson12 committed:
Add wrappers for shiny data filter

Jeff-Thompson12 committed:
Update NEWS

Jeff-Thompson12 committed:
Add reader readme action

Jeff-Thompson12 committed:
Update README to use wrappers

Jeff-Thompson12 committed:
Re-build Rmarkdown files

Jeff-Thompson12 committed:
Clean up comments and examples a little

Jeff-Thompson12 committed:
Merge branch 'jt-22-utilize_moduleServer' of https://github.com/Biogen-Inc/IDEAFilter into jt-22-utilize_moduleServer

Jeff-Thompson12 committed:
Re-build Rmarkdown files

Jeff-Thompson12 committed:
Separate out IDEAFilter modules

Jeff-Thompson12 committed:
Add new filters using modern module implementation

Jeff-Thompson12 committed:
Add tests for new implementation

Jeff-Thompson12 committed:
Update example apps

Jeff-Thompson12 committed:
Fix bug when changing dataset

Jeff-Thompson12 committed:
Repair `data_call`

Jeff-Thompson12 committed:
Repair bug when switching datasets with added filters

Jeff-Thompson12 committed:
Update documentation

Jeff-Thompson12 committed:
Add tests for reactive data

Jeff-Thompson12 committed:
Add waiter

Jeff-Thompson12 committed:
Update NEWS

Jeff-Thompson12 committed:
Merge pull request #23 from Biogen-Inc/jt-22-utilize_moduleServer

Utilize `moduleServer()`

Jeff-Thompson12 committed:
Merge branch 'devel' into jt-13-preselection

Jeff-Thompson12 committed:
Implement pre-selection using `IDEAFilter()`

Jeff-Thompson12 committed:
Increment version number

Jeff-Thompson12 committed:
Implement column subsetting

Jeff-Thompson12 committed:
Update NEWS

Jeff-Thompson12 committed:
Update documentation

Jeff-Thompson12 committed:
Merge pull request #24 from Biogen-Inc/jt-14-column_subsetting

Implement Column Subsetting

Jeff-Thompson12 committed:
Use `pillar::type_sum()` instead of `pillar::new_pillar_type()`

Jeff-Thompson12 committed:
Update NEWS

Jeff-Thompson12 committed:
Merge pull request #25 from Biogen-Inc/jt-9-use_type_sum

Use `pillar::type_sum()`

Jeff-Thompson12 committed:
Merge branch 'devel' into jt-13-preselection

Jeff-Thompson12 committed:
Fix merge conflicts with `dev`

Merge branch 'jt-13-preselection' of https://github.com/Biogen-Inc/IDEAFilter into jt-13-preselection

# Conflicts:
#	DESCRIPTION

Jeff-Thompson12 committed:
Fix bug with NAs imcorrectly displaying

Jeff-Thompson12 committed:
Fix issue where `IDEAFilter()` was pointing to `shiny_data_filter_item_ui()`

Jeff-Thompson12 committed:
Simplify `code()` reactive in `IDEAFilter()`

Jeff-Thompson12 committed:
Add test for pre-selection

Jeff-Thompson12 committed:
Increment version number

Jeff-Thompson12 committed:
Merge pull request #20 from Biogen-Inc/jt-13-preselection

Allow selection of vectors and filters upon initialization

Jeff-Thompson12 committed:
Initialize vignettes

Jeff-Thompson12 committed:
Don't reinitialize modules on resort

Jeff-Thompson12 committed:
Repair test for IDEAFilter_item

Jeff-Thompson12 committed:
Add erase filter selection feature

Jeff-Thompson12 committed:
Remove inputs when exiting module

Jeff-Thompson12 committed:
Remove observers when exiting vector module

Jeff-Thompson12 committed:
Increment version number

Jeff-Thompson12 committed:
Add data to test data types

Jeff-Thompson12 committed:
Fix bug causing the erase filters event to trigger

Jeff-Thompson12 committed:
Update NEWS

Jeff-Thompson12 committed:
Update NEWS again

Jeff-Thompson12 committed:
Merge pull request #26 from Biogen-Inc/jt-vector_filter

Enhance vector filter

Jeff-Thompson12 committed:
Require data

Jeff-Thompson12 committed:
Add checks for data being present

Jeff-Thompson12 committed:
Increment version number

Jeff-Thompson12 committed:
Merge pull request #27 from Biogen-Inc/jt-test_tidyCDISC

Test on `tidyCDISC`

Jeff-Thompson12 commented:
@AARON-CLARK I think we are in a pretty good place here. A lot of improvements and enhancements. Is there anything I'm missing?

I haven't written any vignettes. The README is pretty expansive of the functionality. I suppose I should write up a vignette highlighting some of the new features.

Also, feel free to explore the updated filter on [tidyCDISC](https://github.com/Biogen-Inc/tidyCDISC/tree/jt-test_IDEAFilter).

AARON-CLARK commented:
> I haven't written any vignettes. The README is pretty expansive of the functionality. I suppose I should write up a vignette highlighting some of the new features.

HI @Jeff-Thompson12, I haven't reviewed this in it's entirety (yet), but yes, I would argue for having at least one vignette so we can get credit via `riskmetric` (#6).

Jeff-Thompson12 commented:
I will add a vignette but want to wait for your more complete feedback first.

AARON-CLARK commented:
I love the new eraser option implemented recently. Specifically, that `IDEAFilter` keeps the variable selected, but allows the user to quickly toggle the filter off & restart. Genius!

Not a huge deal, but one thing I would suggest for the reviewer and our future selves is including more information in your PRs so it's easier to identify what the old PRs hoped to achieve - maybe throw in a screenshot. I think that's just "nice to have" so I (and future you) can quickly understand what each PR did/ does. 

Also, examples on how to implement new arguments would be nice, although that info is probably planned for the vignette you want to compose. The column subsetting did not have an example to follow. I would advise adding examples on perfroming both "column subsetting" and "pre-selection" in one or both of the example apps in `inst/`, plus via the `@examples` tag in the `roxygen2`.

FYI, I did turn your `tidyCDISC` branch into an official PR [here](https://github.com/Biogen-Inc/tidyCDISC/pull/247#pullrequestreview-1587457909).

I was able screw some things up during my testing!

## Conflicting col_subset & preselection
First, I was able to crash the STARWARS example app by pre-selecting a variable `name` below, but then removing it using the col_subset feature. 
```r
filtered_data <- IDEAFilter("data_filter",
                             data = starwars2,
                             preselection = list(name = list(filter_na = TRUE,
                                                             filter_fn = ~stringr::str_detect(.x, '[0-9]'))),
                             col_subset = c("height", "mass", "hair_color", "eye_color", "vehicles", "is_droid"),
                             verbose = FALSE)
```
Note, the app didn't crash until I clicked this little button: 

![image](https://github.com/Biogen-Inc/IDEAFilter/assets/17878953/74ebd8d7-dafa-403c-9244-7c8adc6f9045)

The console just displayed:

![image](https://github.com/Biogen-Inc/IDEAFilter/assets/17878953/774ebda3-0423-47b9-be97-99b07308b2a8)

## Failed render when shiny vector type changes
This problem probably existed before your changes, so if you want to push it down the road, that's fine but this was the sequence of events that I tested to reproduce the issue:

For the STARWARS app, select one name and one height. Note that height is a quantitative measure which will show up as a range slider when there exists more data points... I think more than 5 or 8? I can't really remember.

![image](https://github.com/Biogen-Inc/IDEAFilter/assets/17878953/d5633e52-0210-4628-8b59-e97deddafd7c)

So from here, I can do three things to make things go haywire:
1.  de-select Ackbar
2. click the new eraser button
3. shift the height filter to position # 1

I'll do the latter. You can see that the name variable's filter item freaks out and the quantitative height variable adopts a < 180 approach.

![image](https://github.com/Biogen-Inc/IDEAFilter/assets/17878953/58cefd75-22b2-4362-b537-f645e44eabba)


I think the desired result is to have the quantitative filter stick to one number, as shown below:

![image](https://github.com/Biogen-Inc/IDEAFilter/assets/17878953/e8261978-4ac9-4435-8ca4-e7c5505209df)

I think exploring behaviors of these shifting UIs should be explored more deeply, since they exist for other data types as well, not just the quantitative range selectors.

Jeff-Thompson12 commented:
@AARON-CLARK Neither issue predates me.

1. Conflicting `col_subset` and `preselection`
I definitely was the one to introduce this issue. I added [code](https://github.com/Biogen-Inc/IDEAFilter/blob/a917cd33539efdc06dafc5713582951ffb44c596/R/IDEAFilter_item.R#L173-L183) to destroy observers when removing a filter item or changing the column. I also did this for `shiny_data_filter_item()` because of changing made to vector modules. I think we can go one of two ways: (1) make sure the preselection column is in the subset of columns (easy fix) or (2) allow incorrect pre-selections and fix the bug causing the app to crash. My vote is for option 2 because there exists the concept of a NULL column type. This also handles the issue of the pre-selection failing silently.
2. Failed render when shiny vector type changes
Fairly straightforward fix. The issue I found was that the `input$param` is being shared. I added a function `remove_shiny_inputs()` that will take care of this issue when switching the column. This is the only instance were the data type can change in a meaningful way that impacts `input$param` (the inputs for factor few and many are interchangeable). I will push a commit to address this.

Jeff-Thompson12 committed:
Fix bug when switching between numeric few and many

Jeff-Thompson12 committed:
Add `try()` when destroying observer

AARON-CLARK commented:
> 1. My vote is for option 2

 I was just thinking option 1, but if option 2 floats your boat, your boat should float 👍🏼.

AARON-CLARK commented:
Per our discussion today, I found this little issue, likely having to do with the merge between ADSL & ADLB. This exists on `tidyCDISC`'s master branch too.

![image](https://github.com/Biogen-Inc/IDEAFilter/assets/17878953/c8cb9255-72e9-4ba3-a896-b5c1b9074e6c).

Jeff-Thompson12 committed:
Fix merge conflicts with `devel`

Merge branch 'devel' into jt-6-add_vignettes

# Conflicts:
#	DESCRIPTION

Jeff-Thompson12 committed:
Update version number

Jeff-Thompson12 committed:
Resolve minor bug when switching from character vector

Jeff-Thompson12 committed:
Update authorship

Jeff-Thompson12 committed:
Separate inputs in numeric few and many

Jeff-Thompson12 committed:
Update example apps

Jeff-Thompson12 committed:
Flesh out vignette a little more

Jeff-Thompson12 committed:
Change implementation of column subsetting

Jeff-Thompson12 committed:
Increment version number

Jeff-Thompson12 committed:
Update `columnSelectInput()`

Jeff-Thompson12 committed:
Merge pull request #28 from Biogen-Inc/jt-extend_col_subset

Extend column subsetting

Jeff-Thompson12 committed:
Merge branch 'devel' into jt-6-add_vignettes

Jeff-Thompson12 committed:
Increment version number

Jeff-Thompson12 committed:
Re-build Rmarkdown files

Jeff-Thompson12 committed:
Repair tests

Jeff-Thompson12 committed:
Resolve some warnings

AARON-CLARK commented:
@Jeff-Thompson12, was there anything left to do for the next release besides adding the vignette (#29)?

Jeff-Thompson12 commented:
I think that was it. I have a local branch that isn't finished where I was working on that. There were a few changes that I made on that branch as well. I will open a PR for it.

Oops. PR was already opened. #29

Jeff-Thompson12 committed:
Resolve issue where column subsets were not getting updated

Jeff-Thompson12 committed:
Flesh out vignette a little more

Jeff-Thompson12 committed:
Update some language in vignette

Jeff-Thompson12 committed:
Merge pull request #29 from Biogen-Inc/jt-6-add_vignettes

Add a vignette

Jeff-Thompson12 commented:
@AARON-CLARK I think we are there. If you want to take a look at the changes or have any additional checks or edits.

Also, I updated by contribution level from contributor to author. Seemed appropriate to me.

jthompson-arcus committed:
Fix some issues introduced by changes in selectize.js

jthompson-arcus committed:
Fix minor bug applying NA filter to logical vector

jthompson-arcus committed:
Don't show deprecated message for `shiny_data_filter_item` when called inside `shiny_data_filer`

jthompson-arcus committed:
Polish NEWS

jthompson-arcus committed:
Fix URL redirect

jthompson-arcus committed:
Update CRAN comments

jthompson-arcus committed:
Increment for minor release

AARON-CLARK commented on cran-comments.md:
```diff
@@ -1,75 +1,16 @@
+## Reverse dependency check
 
-There are none.
+The 1 reverse dependency was checked by running R CMD check using the development version of this package
```
```suggestion
One reverse dependency exists (`{tidyCDISC}`) and was tested by running R CMD Check using the development version of `IDEAFilter`. The changes have no negative impact on it's reverse dependency.
```

jthompson-arcus committed:
Update cran-comments.md

Co-authored-by: Aaron Clark <CLARK.AARONCHRIS@GMAIL.COM>

aclark02-arcus committed:
.Rbuildignore, description, CRAN-SUB file changed

tidyCDISC committed:
Update LICENSE.md

tidyCDISC committed:
Update LICENSE

tidyCDISC committed:
Update README.Rmd

aclark02-arcus committed:
re-compile readme.md

aclark02-arcus committed:
push updated cran-comments

aclark02-arcus committed:
Add cran submission file

tidyCDISC committed:
Update DESCRIPTION's reference to license

aclark02-arcus committed:
update wordlist & fix spelling

aclark02-arcus committed:
hide spellcheck code

aclark02-arcus committed:
update another spot in readme where link was getting redirected

aclark02-arcus committed:
Merge pull request #30 from Biogen-Inc/prep_cran_020

v0.2.0 for CRAN
